src/bindings/text-buffer-wrapper.cc

```diff
@@ -555,14 +589,20 @@ void TextBufferWrapper::find_words_with_subsequence_in_range(const Nan::Function
   auto callback = new Nan::Callback(info[4].As<Function>());
 
   if (query && extra_word_characters && max_count && range && callback) {
-    Nan::AsyncQueueWorker(new FindWordsWithSubsequenceInRangeWorker(
-      info.This(),
+    auto js_buffer = info.This();
+    auto text_buffer_wrapper = Nan::ObjectWrap::Unwrap<TextBufferWrapper>(js_buffer);
+
+    auto worker = new FindWordsWithSubsequenceInRangeWorker(
+      js_buffer,
       callback,
       *query,
       *extra_word_characters,
       *max_count,
       *range
-    ));
+    );
+
+    text_buffer_wrapper->outstanding_workers.insert(worker);
```

Comment:
I know this is the only case where we have outstanding workers, but might it make sense to give this set and the cancellation method a specific name... to cancel outstanding subsequence searches?

Reply:
I think that in the future, it needs to other types of workers as well. For example, you can trigger the same problem by calling `findAll(regex)` many times rapidly, interspersed with buffer changes. We haven't made those calls cancellable yet. I'm not sure if we should do that as part of this PR, since currently there's nothing in Atom that would call that API on every keystroke.

Reply:
Yeah, that makes sense. This set is for anything that should be cancelled on the next edit.